# origin: akpm (BitKeeper)
# cset: 1.2087 (2.6) key=4233d1160KRwGtYr32RaehQWJRFT_g
# URL: http://linux.bkbits.net:8080/linux-2.6/cset@4233d1160KRwGtYr32RaehQWJRFT_g
# inclusion: upstream
# descrition: [PATCH] cramfs: small stat(2) fix
# revision date: Fri, 08 Apr 2005 14:50:21 +0900
#
# S rset: ChangeSet|1.2086..1.2087
# R rset: fs/cramfs/inode.c|1.38..1.39
#
# Key:
# S: Skipped  ChangeSet file only
# O: Original Followed by Updated
# U: Updated  Included with updated range of versions
# I: Included Included verbatim
# E: Excluded Excluded on request from user
# D: Deleted  Manually deleted by subsequent user edit
# R: Revised  Manually revised by subsequent user edit
#
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/03/12 21:35:18-08:00 akpm@osdl.org 
#   [PATCH] cramfs: small stat(2) fix
#   
#   From: Eric Lammerts <eric@lammerts.org>
#   
#   When I stat(2) a device node on a cramfs, the st_blocks field is bogus
#   (it's derived from the size field which in this case holds the major/minor
#   numbers).  This makes du(1) output completely wrong.
#   
#   Signed-off-by: Eric Lammerts <eric@lammerts.org>
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
# 
# fs/cramfs/inode.c
#   2005/03/04 13:15:57-08:00 akpm@osdl.org +1 -0
#   cramfs: small stat(2) fix
# 
#
===== fs/cramfs/inode.c 1.38 vs 1.39 =====
--- a/fs/cramfs/inode.c	2004-04-17 00:39:37 +09:00
+++ b/fs/cramfs/inode.c	2005-03-05 06:15:57 +09:00
@@ -73,6 +73,7 @@ static struct inode *get_cramfs_inode(st
 			inode->i_data.a_ops = &cramfs_aops;
 		} else {
 			inode->i_size = 0;
+			inode->i_blocks = 0;
 			init_special_inode(inode, inode->i_mode, cramfs_inode->size);
 		}
 	}
